packages/core/src/components/Form.tsx

```diff
@@ -195,6 +196,7 @@ export interface FormProps<T = any, S extends StrictRJSFSchema = RJSFSchema, F e
    * to put the second parameter before the first in its translation.
    */
   translateString?: Registry['translateString'];
+  nameGenerator?: NameGeneratorFunction;
```

Comment:
Please document this new prop, here as well as in the `docs` directory. Also, update the `v6x upgrade guide.md` to mention the new feature (you can point to the docs for it to make it easy)